### PR TITLE
airlocks and computers are leanable

### PR DESCRIPTION
**Pull request by grungussuss**
## About The Pull Request
- added a signal - `COMSIG_ATOM_DENSITY_CHANGED`, is sent when set_density is called and doesn't early return
- changed all manual assignments of .density to use `set_density` so the signal is sent
- airlocks get the leanable component, if the airlock opens while you are leaning on it - you will fall
- computers get the leanable component
## Why It's Good For The Game
can lean on more stuff, roleplay! Immersion!
*John Tider leans on the airlock*
## Changelog
:cl: grungussuss
add: computers and airlocks are now leanable
refactor: changed how density/collision of some objects is changed, report any oddities!
/:cl:

### CODE REVIEW

**Comment by norsvenska** [comment]
you should fall down if the airlock is opened while you're leaning on it

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -182,6 +184,11 @@
 		randomize_language_if_on_station()
 	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_NEW_MACHINE, src)
 
+	/// For performance reasons and to cut down on init times we are "lazy-loading" the leaning component when someone drags their sprite onto us, and then calling dragging code again to trigger the component
```
Comment (MrMelbert) [inline code/game/machinery/_machinery.dm:187]: This comment seems like lies

**Comment by MrMelbert** [comment]
I don't think this was tested actually

**Comment by Sadboysuss** [comment]
Oh actually I forgot to commit all my changes and this isn't the entire thing

File: code/game/machinery/_machinery.dm (hunk 1 above)
Comment (Sadboysuss) [reply code/game/machinery/_machinery.dm]: oh because it's old, see new code

**Comment by Sadboysuss** [comment]
https://github.com/user-attachments/assets/f269d8b9-0d25-4916-af28-650e468370a7

**Review by SyncIt21 — CHANGES_REQUESTED** [review]
Leaning code has slightly changed since #88494

**Comment by Sadboysuss** [comment]
> Leaning code has slightly changed since #88494

I see, thanks for helping adapt it to your changes

**Comment by Sadboysuss** [comment]
I'm gonna test this later, when I do - I'll take it out of draft

**Comment by Sadboysuss** [comment]
it works

**Review by Ghommie — APPROVED** [review]

**Comment by Ghommie** [comment]
Conflicts.

**Pull request merged**